### DISCUSSION TITLE
Weird behavior for panos_security_rule module

### DISCUSSION THREAD

**Discussion by fersherlsgithub** [PaloAltoNetworks/pan-os-ansible#430] [Q&A]
Hey Everyone

I noticed something really interesting when using the **panos_security_rule** module

For some reason, When I try to configure rules in a Palo Alto VM-500 using the Public IP of the mgmt Interface, it works like a charm configures everything as expected.

But If I use a private IP(Same palo alto interface just using the private IP), only the panos_security_rules does not work, it just hangs for a while and throws the following error.

**An exception occurred during task execution. To see the full traceback, use -vvv. The error was: http.client.RemoteDisconnected: Remote end closed connection without response**

Any other playbooks module like for example create_objects, custom_urls, etc. works with the private IP, So I pretty much rule out a network/connectivity/authentication issue with that.

Tried chatgpt since I dont have a good dev background.

it says that it might be a "TypeError" within panos/base.py

Just wondering if there is any ideas about what could be hapenning

**Comment by jamesholland-uk** [comment]
Tracking in #431
